woocommerce/woocommerce-ios
# [Mobile Payments] Add "Collect Payment" button to order details view

Closes #3826 

⚠️ This PR is against the feature branch implementing support for Card Present Payments, not against develop ⚠️
⚠️ Please note there are several parts of the implementation that are not production ready.⚠️

We continue building our prototype (for reference, p91TBi-4q4-p2), this time, by implementing logic to display a "Collect Payment" button in the order details view.

For a non elegible order:
<img src="https://user-images.githubusercontent.com/2722505/111922517-9e325100-8a70-11eb-8787-f7fe87353974.PNG" width="350"/>

And for an elegible order:

| Light appearance | Dark appearance |
| ------ | ------ |
| <img src="https://user-images.githubusercontent.com/2722505/111922548-d5a0fd80-8a70-11eb-9e50-59a682009c4a.PNG" width="350"/> | <img src="https://user-images.githubusercontent.com/2722505/111922562-eea9ae80-8a70-11eb-81db-1b46e6c3b623.PNG" width="350"/> |


## Changes
* Update OrderDetalsViewController, OrderDetailsViewModel and OrderDetailsDataSource, to add a "Collect Payment" button that should be visible only on orders with "Cash On Delivery" as payment method.

## Testing
⚠️ Tapping the button will crash the app, unless a card reader has been paired first. This is expected (for now) ⚠️

* Checkout the branch.
* Run `bundle exec pod install`
* Navigate to an order with payment method "cash on delivery". 
* Notice the button labeled "Collect Payment" in the payment section.
* If testing with a hardware reader, connect to a reader first (Settings> Manage Card Readers > Connect card reader and follow the prompts) Then navigate back to the order and tap Collect Payment. That should trigger the payment collection flow

Update release notes:
